include/pwm.h

```diff
@@ -70,6 +70,8 @@ struct pwm_init_param {
 	uint32_t period_ns;
 	/** PWM generator duty cycle */
 	uint32_t duty_cycle_ns;
+	/** PWM generator offset */
```

Comment:
Maybe you can add more in the comment about what the offset is, I'am just understood reading the commit description.

Reply:
Anyway. Isn't this a feature that should be in extra?
For example, an MCU can generate a PMW using timers, I don't know if there would be a reliable way to synchronize all the timers to have a desired offset between different PWMs.

Reply:
If the controller supports polarity, you can use for example inverted polarity and the duty cycle width will represent a time offset relative to the moment when the pwm was enabled

Reply:
Or for example you are using two timers to generate an interrupt based pwm. If they both start counting from 0, the rising edge will be alligned, but if you set a different value inside the first timer's counter, you will get a phase shift between the signals

Reply:
I'm also a bit confused, is this a phase offset ? If yes, you could write it like that, then it's much more clear

Reply:
I agree, the 'offset' naming came out of the hdl registers. Will change